tests/unit/amo/components/TestAddon.js

```diff
@@ -91,11 +118,123 @@ describe('Addon', () => {
     sinon.assert.calledWith(fakeDispatch, setViewContext(fakeAddon.type));
   });
 
+  it('dispatches setViewContext on update', () => {
+    const fakeDispatch = sinon.stub();
+    const root = render({ dispatch: fakeDispatch });
+    fakeDispatch.reset();
+    root.componentWillReceiveProps({
+      addon: { ...fakeAddon, type: ADDON_TYPE_THEME },
+    });
+
+    sinon.assert.calledWith(
+      fakeDispatch, setViewContext(ADDON_TYPE_THEME));
+  });
+
+  it('dispatches setViewContext when updating with new addon', () => {
+    const fakeDispatch = sinon.stub();
+    // Start with a null addon
+    const root = render({
+      addon: null, dispatch: fakeDispatch, params: { slug: 'some-slug' },
+    });
+    fakeDispatch.reset();
+    // Update with a new addon
+    root.componentWillReceiveProps({ addon: fakeAddon });
+
+    sinon.assert.calledWith(fakeDispatch, setViewContext(fakeAddon.type));
+  });
+
+  it('only dispatches setViewContext for a new addon type', () => {
+    const fakeDispatch = sinon.stub();
+    const root = render({ addon: fakeAddon, dispatch: fakeDispatch });
+    fakeDispatch.reset();
+    // Update with the same addon (this apparently happens IRL).
```

Comment:
Ah, weird. I think `componentDidUpdate` is only called if the props changed whereas `componentWillUpdate` is called either way maybe? My bad if I suggested the wrong one. We should double-check we're using the right one according to the lifecycle docs: https://facebook.github.io/react/docs/react-component.html#the-component-lifecycle